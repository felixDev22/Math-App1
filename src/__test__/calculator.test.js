import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import CalculatorUI from '../components/calculator';

describe('testing calculator component with user interaction', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CalculatorUI />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should calculate the value', () => {
    render(<CalculatorUI />);
    const btn1 = screen.getByRole('button', { name: /1/i });
    userEvent.click(btn1);
    const btnOperation = screen.getByRole('button', { name: /+/i });
    userEvent.click(btnOperation);
    const btn2 = screen.getByRole('button', { name: /5/i });
    userEvent.click(btn2);
    const btnEqualto = screen.getByRole('button', { name: /=/i });
    userEvent.click(btnEqualto);
    const result = screen.getByTestId('total_value');
    expect(result.testContent).toBe('6');
  });
});
