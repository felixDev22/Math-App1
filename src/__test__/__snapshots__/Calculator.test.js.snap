// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cal-section"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="cal"
  >
    <div
      className="displayArea"
    >
      <p
        className="cal-input"
        type="text"
      >
        0
      </p>
    </div>
    <div
      className="operators"
    >
      <ul
        className="rows"
      >
        <li
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="btn-2"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </li>
        <li
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn-2"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </li>
        <li
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn-2"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </li>
        <li
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn-2"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </li>
        <li
          className="row2"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="btn-2"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
